docs/content/stable/develop/quality-of-service/write-heavy-workloads.md

```diff
@@ -23,66 +24,59 @@ The idea is to slow down incoming writes to the speed that the database can hand
 ## Identifying reasons for write stalls
 
 Write stalls can occur for the following reasons:
+
 * Low CPU environments
 * Disks with low performance
 
 The following symptoms occur at the database layer:
 
 ### Flushes cannot keep up
+
 More writes than the system can handle could result in too many memtables getting created, which are queued up to get flushed. This puts the system in a suboptimal state, because a failure would need a large rebuild of data from WAL files, as well as requiring too many compactions for the system to catch up to a healthy state.
 
 ### Compaction cannot keep up
+
 The database getting overloaded with writes can also result in compactions not being able to keep up. This causes the SST files to pile up, degrading read performance significantly.
 
 ### WAL writes are too slow
-If WAL writes are slow, then writes experience higher latency which creates a natural form of admission control. The frequency of data getting synced to disk is controlled by the `durable_wal_write` flag. Note that fsync is disabled by default since YugabyteDB is expected to run in a Replication Factor 3 mode where each shard/tablet is replicated onto 3 independent fault domains such as hosts, availability zones, regions or clouds. If enabled, every single write into DocDB, YugabyteDB’s underlying document store, will be synchronized to disk before its execution is deemed successful. So, although there will be an increase in safety, there will be a hit to performance.
+
+If WAL writes are slow, then writes experience higher latency which creates a natural form of admission control. The frequency of data getting synced to disk is controlled by the `durable_wal_write` flag. Note that `fsync` is disabled by default as YugabyteDB is expected to run in a Replication Factor 3 mode where each shard/tablet is replicated onto 3 independent fault domains such as hosts, availability zones, regions, or clouds. If enabled, every single write into DocDB, YugabyteDB's underlying document store, will be synchronized to disk before its execution is deemed successful. So, although there will be an increase in safety, there will be a hit to performance.
 
 ### Limited disk IOPS or bandwidth
-In many cloud environments, the disk IOPS and/or the network bandwidth is rate-throttled. Holistically, if there are such disk constraints (limits on iops / bandwidth), that results in a back-pressure across all disk writes in the system, and would therefore manifest as one of the above conditions.
 
+In many cloud environments, the disk IOPS and/or the network bandwidth is rate-throttled. Holistically, if there are such disk constraints (limits on IOPS / bandwidth), that results in a back-pressure across all disk writes in the system, and would therefore manifest as one of the above conditions.
```

Comment:
In many cloud environments, the disk IOPS and network bandwidth are rate-throttled. Holistically, if there are such disk constraints (limits on IOPS and bandwidth), it results in a back-pressure across all disk writes in the system and would manifest as one of the preceding conditions.